### PR TITLE
Comments for review and discussion

### PR DESCRIPTION
**Pull request by nickrandolph**

### CODE REVIEW

File: device_calendar/lib/src/device_calendar.dart [hunk 1]
```diff
@@ -74,6 +81,8 @@ class DeviceCalendarPlugin {
       res.errorMessages.add(
           "[${ErrorCodes.invalidArguments}] ${ErrorMessages.invalidMissingCalendarId}");
     }
+
+    // TODO: Extend capability to handle null start or null end (eg all events after a certain date (null end date) or all events prior to a certain date (null start date))
```
Comment (MaikuB) [inline device_calendar/lib/src/device_calendar.dart]: iOS requires both start and end to be specified if dates are used and this is the only predicate that they allow using
https://developer.apple.com/documentation/eventkit/ekeventstore/1507479-predicateforeventswithstartdate?language=objc

retrieve by id is done using this https://developer.apple.com/documentation/eventkit/ekeventstore/1507490-eventwithidentifier?language=objc

File: device_calendar/lib/src/common/error_codes.dart [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class ErrorCodes {
   static const int invalidArguments = 400;
+  // TODO: Add error code 500 for when we get an error back from the device level
```
Comment (mkieres) [inline device_calendar/lib/src/common/error_codes.dart]: Let's add platform exception (500)

File: device_calendar/lib/src/device_calendar.dart [hunk 3]
```diff
@@ -16,21 +16,26 @@ class DeviceCalendarPlugin {
   DeviceCalendarPlugin._createInstance();
 
   /// Requests permissions to modify the calendars on the device
+  /// TODO: add comment about return value
```
Comment (mkieres) [inline device_calendar/lib/src/device_calendar.dart]: Investigate on how to create comments on the code - input parameters and return values

File: device_calendar/lib/src/models/result.dart [hunk 4]
```diff
@@ -5,5 +5,11 @@ class Result<T> {
   T data;
   List<String> errorMessages = new List<String>();
 
+    // TODO: Change to allow constructor parameter
   Result([this.data]);
+
```
Comment (mkieres) [inline device_calendar/lib/src/models/result.dart]: 1. Make IsSuccess flag dependant on Data not being null and error messages being empty
2. Change constructor to be parameter free

File: device_calendar/lib/src/device_calendar.dart [hunk 5]
```diff
@@ -16,21 +16,26 @@ class DeviceCalendarPlugin {
...
 
     try {
       res.data = await channel.invokeMethod('requestPermissions');
+      // TODO: Remove calls to set isSuccess to true
       res.isSuccess = true;
-    } on PlatformException catch (e) {
+    } on PlatformException catch (e) { 
+      // TODO: Change parse method to return a new Result that can be returned immediately     
       _parsePlatformExceptionAndUpdateResult<bool>(e, res);
+      // TODO: Move print into the parse method
```
Comment (mkieres) [inline device_calendar/lib/src/device_calendar.dart]: Move this to the parsing method

File: device_calendar/lib/src/device_calendar.dart (hunk 1 above)
Comment (mkieres) [reply device_calendar/lib/src/device_calendar.dart]: Pushing this to the next iteration

File: device_calendar/lib/src/device_calendar.dart [hunk 6]
```diff
@@ -85,6 +94,7 @@ class DeviceCalendarPlugin {
           "[${ErrorCodes.invalidArguments}] ${ErrorMessages.invalidRetrieveEventsParams}");
     }
 
+    // TODO: Change this to check Success
```
Comment (mkieres) [inline device_calendar/lib/src/device_calendar.dart]: Doing it!

File: device_calendar/lib/src/device_calendar.dart [hunk 7]
```diff
@@ -168,9 +180,11 @@ class DeviceCalendarPlugin {
     return res;
   }
 
+  // TODO: Change this to return a new Result based on the parsed exception
```
Comment (mkieres) [inline device_calendar/lib/src/device_calendar.dart]: Handle scenario when the exception is null - create a generic message about 'unknown' platform error

**Pull request merged**